cypress/e2e/codeReview/codeReview.cy.ts

```diff
@@ -321,3 +324,105 @@ describe('Comment settings', () => {
     );
   });
 });
+
+describe('Show Line Numbers Setting Builder View', () => {
+  beforeEach(() => {
+    cy.setUpApi({
+      database: {
+        appSettings: [
+          MOCK_GENERAL_SETTINGS,
+          {
+            ...MOCK_GENERAL_SETTINGS,
+            data: {
+              ...DEFAULT_GENERAL_SETTINGS,
+              [GeneralSettingsKeys.ShowLineNumbers]: true,
+            },
+          },
+          MOCK_CODE_SETTINGS,
+        ],
+      },
+      appContext: {
+        context: Context.BUILDER,
+        permission: PermissionLevel.Admin,
+      },
+    });
+    cy.visit('/');
+  });
+
+  it('Toggling Line Numbers Switch from Display Settings', () => {
+    // Open Dispalay Settings.
+    cy.get(buildDataCy(DISPLAY_SETTINGS_FAB_CYPRESS))
+      .should('be.visible')
+      .as('displaySettingsFab')
+      .click();
+    // Check that Line Numbers Switch is checked.
+    cy.get(`[data-cy=${SHOW_LINE_NUMBERS_SWITCH_CYPRESS}]`).within(() => {
+      cy.get('input[type="checkbox"]').should('be.checked');
+    });
+    // Check that line numbers are shown.
+    cy.get('[data-cy=line-number]').should('exist');
+    // Click on switch to toggle setting, check that the switch shouldn't be checked.
+    cy.get(`[data-cy=${SHOW_LINE_NUMBERS_SWITCH_CYPRESS}]`).within(() => {
+      cy.get('input[type="checkbox"]').click();
+      cy.get('input[type="checkbox"]').should('not.be.checked');
+    });
+    // Click on Save button from the display settings window.
+    cy.get(buildDataCy(SETTINGS_DIALOG_SAVE_BUTTON_CYPRESS))
+      .as('saveButton')
+      .should('be.visible')
+      .click();
+    // Check that line numbers are not shown.
+    cy.get('[data-cy=line-number]').should('not.exist');
+  });
+});
+
+describe('Show Line Numbers Setting Player View', () => {
+  beforeEach(() => {
+    cy.setUpApi({
+      database: {
+        appSettings: [
+          MOCK_GENERAL_SETTINGS,
+          {
+            ...MOCK_GENERAL_SETTINGS,
+            data: {
+              ...DEFAULT_GENERAL_SETTINGS,
+              [GeneralSettingsKeys.ShowLineNumbers]: true,
+            },
+          },
+          MOCK_CODE_SETTINGS,
+        ],
+      },
+      appContext: {
+        context: Context.PLAYER,
+        permission: PermissionLevel.Read,
+      },
+    });
+    cy.visit('/');
+  });
+  it('Should display line numbers when "ShowLineNumbers" is ture', () => {
+    cy.get('[data-cy=line-number]').should('not.exist');
```

Comment:
Same comment about extracting this to `selectors.ts` and using the utility function.